app/models/container_service.rb

```diff
@@ -6,7 +6,7 @@ class ContainerService < ApplicationRecord
 
   belongs_to  :ext_management_system, :foreign_key => "ems_id"
   has_and_belongs_to_many :container_groups, :join_table => :container_groups_container_services
-  has_many :container_routes
+  has_many :container_routes, :dependent => :destroy # TODO: consider nullify for things like routes, which should be removed by event handling
```

Comment:
Routes are a top-level openshift collection that should be deleted by the refresher